collated/main/lawwman.md

```diff
@@ -0,0 +1,888 @@
+    }
+
+```
+###### \java\seedu\address\logic\parser\ParserUtil.java
+``` java
+    /**
+     * Parses a {@code Optional<String> deadline} into an {@code Optional<Deadline>} if {@code Deadline}
+     * is present.
+     * Meant for parsing for Edit command.
+     * See header comment of this class regarding the use of {@code Optional} parameters.
+     */
+    public static Optional<Deadline> parseDeadlineForEdit(Optional<String> deadline)
+            throws IllegalValueException {
+        requireNonNull(deadline);
+        return deadline.isPresent() ? Optional.of(new Deadline(deadline.get())) : Optional.empty();
+    }
+```
+###### \java\seedu\address\logic\parser\ParserUtil.java
+``` java
+    /**
+     * Parses a {@code Optional<String> interest} into an {@code Optional<Interest>} if {@code Interest}
+     * is present.
+     * Meant for parsing for Add command.
+     * See header comment of this class regarding the use of {@code Optional} parameters.
+     */
+    public static Optional<Interest> parseInterest(Optional<String> interest) throws IllegalValueException {
+        requireNonNull(interest);
+        return interest.isPresent() ? Optional.of(new Interest(interest.get()))
+                : Optional.of(new Interest(Interest.NO_INTEREST_SET));
+    }
+
+```
+###### \java\seedu\address\logic\parser\ParserUtil.java
+``` java
+    /**
+     * Parses a {@code Optional<String> Interest} into an {@code Optional<Interest} if {@code Interest}
+     * is present.
+     * Meant for parsing for Edit command.
+     * See header comment of this class regarding the use of {@code Optional} parameters.
+     */
+    public static Optional<Interest> parseInterestForEdit(Optional<String> interest)
+            throws IllegalValueException {
+        requireNonNull(interest);
+        return interest.isPresent() ? Optional.of(new Interest(interest.get())) : Optional.empty();
+    }
+
+```
+###### \java\seedu\address\model\AddressBook.java
+``` java
+    /**
+     * Adds a person to the overdue debt list in the address book.
+     * @return ReadOnly newOverduePerson
+     */
+    public ReadOnlyPerson addOverdueDebtPerson(ReadOnlyPerson p) {
+        int index;
+        index = persons.getIndexOf(p);
+
+        Person newOverduePerson = new Person(p);
+        newOverduePerson.setHasOverdueDebt(true);
+        try {
+            updatePerson(p, newOverduePerson);
+        } catch (DuplicatePersonException e) {
+            throw new AssertionError("The target person cannot be a duplicate");
+        } catch (PersonNotFoundException e) {
+            throw new AssertionError("This is not possible as prior checks have been done");
+        }
+        return persons.getReadOnlyPerson(index);
+    }
+
+```
+###### \java\seedu\address\model\AddressBook.java
+``` java
+    /**
+     * Updates {@code key} to exclude {@code key} from the overdue list in this {@code AddressBook}.
+     * @return ReadOnly newOverdueDebtPerson
```

Comment:
The description should be more descriptive perhaps instead of `newOverdurDebtPerson`, maybe `a person with debt that is newly overdued`